tchannel/tchannel.py

```diff
@@ -319,3 +321,50 @@ def advertise(self, routers=None, name=None, timeout=None,
         response = Response(json.loads(body), headers or {})
 
         raise gen.Return(response)
+
+    @gen.coroutine
+    def stop(self, reason=None):
+        """Gracefully shut down tchannel.
+        :param reason:
+            User can specify the reason for the stop action.
+        """
+        yield self.drain(reason)
+        self._dep_tchannel.peer_group.clear()
+        self._dep_tchannel._server.stop()
+        self._dep_tchannel._server = None
+        self._dep_tchannel._state = State.ready
+        if self.known_peers:
+            for peer_hostport in self.known_peers:
+                self._dep_tchannel.peer_group.add(peer_hostport)
```

Comment:
Still don't understand why this is happening. `relaunch after stop` should be indistinguishable from a normal `listen()`.